speaker/bgp_controller.go

```diff
@@ -34,53 +38,29 @@ import (
 )
 
 type peer struct {
-	cfg *config.Peer
-	bgp session
+	Cfg *config.Peer
+	BGP session
 }
 
 type bgpController struct {
-	logger     log.Logger
-	myNode     string
-	nodeLabels labels.Set
-	peers      []*peer
-	svcAds     map[string][]*bgp.Advertisement
+	logger          log.Logger
+	myNode          string
+	nodeAnnotations labels.Set
+	nodeLabels      labels.Set
+	cfg             *config.Config
+	peers           []*peer
+	peersLock       sync.Mutex
+	svcAds          map[string][]*bgp.Advertisement
 }
 
 func (c *bgpController) SetConfig(l log.Logger, cfg *config.Config) error {
-	newPeers := make([]*peer, 0, len(cfg.Peers))
-newPeers:
-	for _, p := range cfg.Peers {
-		for i, ep := range c.peers {
-			if ep == nil {
-				continue
-			}
-			if reflect.DeepEqual(p, ep.cfg) {
-				newPeers = append(newPeers, ep)
-				c.peers[i] = nil
-				continue newPeers
-			}
-		}
-		// No existing peers match, create a new one.
-		newPeers = append(newPeers, &peer{
-			cfg: p,
-		})
+	if reflect.DeepEqual(c.cfg, cfg) {
+		return nil
 	}
 
-	oldPeers := c.peers
-	c.peers = newPeers
-
-	for _, p := range oldPeers {
-		if p == nil {
-			continue
-		}
-		level.Info(l).Log("event", "peerRemoved", "peer", p.cfg.Addr, "reason", "removedFromConfig", "msg", "peer deconfigured, closing BGP session")
-		if p.bgp != nil {
-			if err := p.bgp.Close(); err != nil {
-				level.Error(l).Log("op", "setConfig", "error", err, "peer", p.cfg.Addr, "msg", "failed to shut down BGP session")
-			}
-		}
-	}
+	c.cfg = cfg
```

Comment:
How about storing in controller only cfg.Peers as `configPeers` or something like that?
My understanding is that it's used only for that purpouse